CMakeLists.txt

```diff
@@ -48,19 +48,134 @@ if(POLICY CMP0135)
   cmake_policy(SET CMP0135 NEW)
 endif()
 
+if(WIN32)
+  if(NOT DEFINED BUILDENV_BASEPATH)
+    if(DEFINED ENV{BUILDENV_BASEPATH})
+      set(BUILDENV_BASEPATH "$ENV{BUILDENV_BASEPATH}")
+    else()
+      set(BUILDENV_BASEPATH "${CMAKE_SOURCE_DIR}/buildenv")
+    endif()
+  endif()
+
+  if(NOT DEFINED BUILDENV_URL)
+    if(DEFINED ENV{BUILDENV_URL})
+      set(BUILDENV_URL "$ENV{BUILDENV_URL}")
+    else()
+      message(STATUS "BUILDENV_URL not specified")
+    endif()
+  endif()
+
+  if(DEFINED BUILDENV_URL)
+    # Extract the filename from the URL
+    get_filename_component(FILENAME_WITH_EXTENSION "${BUILDENV_URL}" NAME)
+
+    # Remove the extension from the filename
+    # Note, that our buildenv file names contain two dots, e.g. mixxx-deps-2.5-x64-windows-875f13b.zip
+    string(
+      REGEX REPLACE
+      "\\.[^.]*$"
+      ""
+      BUILDENV_NAME
+      "${FILENAME_WITH_EXTENSION}"
+    )
+  else()
+    if(NOT DEFINED BUILDENV_NAME)
+      if(DEFINED ENV{BUILDENV_NAME})
+        set(BUILDENV_NAME "$ENV{BUILDENV_NAME}")
+      else()
+        message(
+          FATAL_ERROR
+          "Neither BUILDENV_URL nor BUILDENV_NAME specified!\nDid you run `${CMAKE_SOURCE_DIR}/tools/windows_buildenv.bat`?"
+        )
+      endif()
+    endif()
+  endif()
+
+  message(STATUS "BUILDENV_NAME is: ${BUILDENV_NAME}")
+
+  if(
+    EXISTS "${BUILDENV_BASEPATH}/${BUILDENV_NAME}"
+    AND IS_DIRECTORY "${BUILDENV_BASEPATH}/${BUILDENV_NAME}"
+  )
+    message(
+      STATUS
+      "BUILDENV directory ${BUILDENV_BASEPATH}/${BUILDENV_NAME} exists!"
+    )
+  else()
+    if(NOT DEFINED BUILDENV_URL)
+      message(
+        FATAL_ERROR
+        "BUILDENV_URL not specified!\nDid you run `${CMAKE_SOURCE_DIR}/tools/windows_buildenv.bat`?"
+      )
+    endif()
+    if(NOT DEFINED BUILDENV_SHA256)
+      if(DEFINED ENV{BUILDENV_SHA256})
+        set(BUILDENV_SHA256 "$ENV{BUILDENV_SHA256}")
+      else()
+        message(STATUS "BUILDENV_SHA256 not specified")
```

Comment:
What might have happen for this message? Can we stop here fatal?

Reply:
This is intentional, similar to the checksums of VCPKG ports. You can run it with out checksum specified. It will of cause fail than later, but with the actual checksum printed, that you can copy paste than.

Reply:
It looks like the check is skipped below in this case, is it? Maybe it is cleaner to move this check closer to usage, to make it easier to trace the control flow. Do we need a bypass feature?
VCPKG has handled a missing checksum like this: 
```
  vcpkg_download_distfile requires a SHA512 argument.
  If you do not know the SHA512, add it as 'SHA512 0' and retry.
```
do we want to copy that?

Reply:
In the VCPKG case you need to write something into the file (works with any value btw. not only 0). But here this would be an additional step without benefit. currently it prints the information you need to specify BUILDENV_SHA256 without such an extra step. I think better developer experience is not possible here.